src/main/java/net/minestom/server/thread/ThreadProvider.java

```diff
@@ -1,252 +1,301 @@
+                // The thread never had any task
+                countDownLatch.countDown();
+                continue;
+            }
+
+            // Execute tick
+            thread.getMainRunnable().startTick(countDownLatch, () -> {
+                final var entitiesList = chunkEntries.stream().map(chunkEntry -> chunkEntry.entities).collect(Collectors.toList());
+                final var entities = entitiesList.stream()
+                        .flatMap(Collection::stream)
+                        .collect(Collectors.toList());
+                Acquirable.refreshEntities(Collections.unmodifiableList(entities));
+
+                final ReentrantLock lock = thread.getLock();
+                lock.lock();
+                chunkEntries.forEach(chunkEntry -> {
+                    Chunk chunk = chunkEntry.chunk;
+                    if (!ChunkUtils.isLoaded(chunk))
+                        return;
+                    chunk.tick(time);
+                    chunkEntry.entities.forEach(entity -> {
+                        final boolean hasQueue = lock.hasQueuedThreads();
+                        if (hasQueue) {
+                            lock.unlock();
+                            // #acquire callbacks should be called here
+                            lock.lock();
+                        }
+                        entity.tick(time);
+                    });
+                });
+                lock.unlock();
+            });
+        }
+        return countDownLatch;
     }
 
-    /**
-     * Executes an entity tick for object entities in an instance's chunk.
-     *
-     * @param instance the chunk's instance
-     * @param chunk    the chunk
-     * @param time     the current time in ms
-     */
-    protected void updateObjectEntities(@NotNull Instance instance, @NotNull Chunk chunk, long time) {
-        conditionalEntityUpdate(instance, chunk, time, entity -> entity instanceof ObjectEntity);
+    public synchronized void refreshThreads(long tickTime) {
+        // Clear removed entities
+        {
+            for (Entity entity : removedEntities) {
+                Acquirable<Entity> acquirable = entity.getAcquiredElement();
+                ChunkEntry chunkEntry = acquirable.getHandler().getChunkEntry();
+                // Remove from list
+                if (chunkEntry != null) {
+                    chunkEntry.entities.remove(entity);
+                }
+            }
+            this.removedEntities.clear();
+        }
+
+
+        final int timeOffset = MathUtils.clamp((int) ((double) tickTime * refreshPercentage), min, max);
+        final long endTime = System.currentTimeMillis() + timeOffset;
+        final int size = chunks.size();
+        int counter = 0;
+        while (true) {
+            Chunk chunk = chunks.pollFirst();
+            if (!ChunkUtils.isLoaded(chunk)) {
+                removeChunk(chunk);
+                return;
+            }
+
+            // Update chunk threads
+            {
```

Comment:
Why is this scoped?

Reply:
Just an easy way for me to differentiate the different parts of a thread refresh, and to say that this method is independent of the rest